### PR TITLE
Enforce eslint no-explicit-any

### PR DESCRIPTION
**Pull request by Arnei**
Adds more typing, but also adds a few eslint exceptions. I tried to keep the exceptions to types were it makes sense to keep `any` (i.e. third party libraries returning it or just cases were we don't care about the contents of a variable). This may be controversial though.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,8 +4,7 @@ module.exports = {
     "react-app/jest",
   ],
   rules: {
-    // Currently 120 warnings.
-    "@typescript-eslint/no-explicit-any": "off",
+
   },
```
Comment (geichelberger) [inline .eslintrc.js:8]: Does it have to be empty, or can we remove that?

File: src/redux/subtitleSlice.ts [hunk 2]
```diff
@@ -252,8 +252,45 @@ export const selectHasChanges = (state: { subtitleState: { hasChanges: subtitle[
...
+//         }
+//       }
+//     }
+
+//     dispatch(setCurrentlyAt(milliseconds));
+//     if (triggered) {
+//       dispatch(setPreviewTriggered(true));
+//     }
+//   };
+// }
```
Comment (geichelberger) [inline src/redux/subtitleSlice.ts:291]: Is that a draft?

**Review by geichelberger — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (Arnei) [reply .eslintrc.js]: Can remove that.

File: src/redux/subtitleSlice.ts (hunk 2 above)
Comment (Arnei) [reply src/redux/subtitleSlice.ts]: Iirc I was testing out function definitions, this can go now.

**Review by geichelberger — APPROVED** [review]
LGTM

**Pull request merged**